### PR TITLE
feat: Salary Table added

### PR DESCRIPTION
**Pull request by 0hmoony**

### CODE REVIEW

**Comment by 0hmoony** [comment]
Screen Shot 2022-08-09 at 12 04 02

File: frontend/package.json [hunk 1]
```diff
@@ -18,9 +18,11 @@
     "@mui/lab": "^5.0.0-alpha.83",
     "@mui/material": "^5.8.1",
     "@mui/x-data-grid": "^5.8.0",
+    "@tanstack/react-table": "^8.5.11",
```
Comment (GereltuyaMz) [inline frontend/package.json:21]: what is this library?

File: frontend/package.json [hunk 2]
```diff
@@ -18,9 +18,11 @@
...
     "@types/uuid": "^8.3.4",
     "aws4-axios": "^2.4.9",
     "axios": "^0.27.2",
+    "bootstrap": "5.2.0",
```
Comment (GereltuyaMz) [inline frontend/package.json:25]: We are currently not using bootstrap

File: frontend/src/components/profile/type.ts [hunk 3]
```diff
@@ -57,3 +57,9 @@ export type AchievementListType = {
   category: string;
   data: Array<AchievementItemType>;
 };
+
+export type SalaryTableType = {
```
Comment (GereltuyaMz) [inline frontend/src/components/profile/type.ts:61]: zadalj bichvel nice to have

File: frontend/src/components/profile/mockData.ts [hunk 4]
```diff
@@ -118,3 +118,60 @@ export const CHART_MOCK_DATA: ChartType = {
     },
   ],
 };
+
```
Comment (GereltuyaMz) [inline frontend/src/components/profile/mockData.ts:121]: single user ?

File: frontend/src/components/profile/salary/index.tsx [hunk 5]
```diff
@@ -121,10 +135,40 @@ const Salary: FC = () => {
               })}
             </Stack>
           </Box>
+          <Box className="salaryTable">
+            <TableContainer component={Paper}>
+              <Table sx={{ minWidth: 650 }} aria-label="simple table">
+                <TableHead>
+                  <TableRow>
+                    <TableCell>Name</TableCell>
```
Comment (billduluu) [inline frontend/src/components/profile/salary/index.tsx:143]: Table Cell deer um nemegdwel yaahu? ingj dawtaj bichihkushuu

File: frontend/src/components/profile/type.ts [hunk 6]
```diff
@@ -57,3 +57,22 @@ export type AchievementListType = {
...
+  {
+    name: string;
+    job: string;
+    salary: number;
+    workdays: number;
+    workedDays: number;
+    bonus: number;
+    calculatedSalary: number;
+    socialSecurityDeduction: number;
+    HOAOT: number;
```
Comment (billduluu) [inline frontend/src/components/profile/type.ts:71]: iim nershil buruu bha

File: frontend/src/components/profile/salary/index.tsx [hunk 7]
```diff
@@ -121,10 +136,29 @@ const Salary: FC = () => {
               })}
             </Stack>
           </Box>
+          <Box className="salaryTable">
+            <TableContainer component={Paper}>
+              <Table sx={{ minWidth: 650 }} aria-label="simple table">
+                <TableHead>
+                  <TableRow sx={{ "&:last-child td, &:last-child th": { border: 0 } }}>
```
Comment (GereltuyaMz) [inline frontend/src/components/profile/salary/index.tsx:143]: would be more visible if add border on the table?

File: frontend/src/components/profile/type.ts [hunk 8]
```diff
@@ -57,3 +57,39 @@ export type AchievementListType = {
...
+    prePayment: number;
+    otherDeduction: number;
+    totalDeduction: number;
+    salaryAtHand: number;
+    card: number;
+  }
+];
+
+export type SalaryTableHeaderType = [
+  string,
```
Comment (GereltuyaMz) [inline frontend/src/components/profile/type.ts:81]:
```suggestion
Array<string>
```

File: frontend/src/components/profile/salary/index.tsx [hunk 9]
```diff
@@ -1,11 +1,23 @@
-import { FC } from "react";
-import { Typography, Grid, Box, Stack } from "@mui/material";
+import { FC, useState } from "react";
```
Comment (GereltuyaMz) [inline frontend/src/components/profile/salary/index.tsx:1]: would be nice to delete the value, if the state unused

File: frontend/src/components/profile/salary/index.tsx [hunk 10]
```diff
@@ -121,10 +136,29 @@ const Salary: FC = () => {
               })}
             </Stack>
           </Box>
+          <Box className="salaryTable">
+            <TableContainer component={Paper}>
+              <Table sx={{ minWidth: 650 }} aria-label="simple table">
```
Comment (GereltuyaMz) [inline frontend/src/components/profile/salary/index.tsx:141]: suggest to delete aria-label if don't know the purpose or change the name to be more suitable for the table

File: frontend/src/components/profile/type.ts [hunk 11]
```diff
@@ -57,3 +57,39 @@ export type AchievementListType = {
...
+    workedDays: number;
+    raise: number;
+    calculatedSalary: number;
+    socialInsurance: number;
+    PIT: number;
+    prePayment: number;
+    otherDeduction: number;
+    totalDeduction: number;
+    salaryAtHand: number;
+    card: number;
```
Comment (GereltuyaMz) [inline frontend/src/components/profile/type.ts:76]: Would the card field usable in salary table?

File: frontend/src/components/profile/salary/index.tsx [hunk 12]
```diff
@@ -121,10 +136,29 @@ const Salary: FC = () => {
...
+                <TableHead>
+                  <TableRow sx={{ "&:last-child td, &:last-child th": { border: 0 } }}>
+                    {TableHeaderData.map(headers => {
+                      return <TableCell align="center">{headers}</TableCell>;
+                    })}
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  <TableRow key={TableDataInfo[0].name} sx={{ "&:last-child td, &:last-child th": { border: 0 } }}>
+                    {Object.keys(TableDataInfo[0]).map(x => {
```
Comment (GereltuyaMz) [inline frontend/src/components/profile/salary/index.tsx:151]: readable name of x?

**Pull request merged**